libraries/griptape_nodes_library/griptape_nodes_library/utils/artifact_path_tethering.py

```
@@ -527,23 +527,53 @@ def _is_url(self, path: str) -> bool:
     def _upload_file_to_static_storage(self, file_path: str) -> str:
         """Upload file to static storage and return download URL."""
         path = Path(file_path)
-        file_name = path.name
+        workspace_path = GriptapeNodes.ConfigManager().workspace_path
+
+        # Resolve to absolute path relative to workspace
+        if not path.is_absolute():
+            path = workspace_path / path
+        else:
+            # If absolute, check if it's within workspace
+            try:
+                path = path.relative_to(workspace_path)
```

Shouldn't swallow exceptions like this. Use `is_relative_to`.